clang/lib/Driver/Driver.cpp

```diff
@@ -5538,6 +5538,17 @@ Action *Driver::ConstructPhaseAction(
       assert(OutputTy != types::TY_INVALID &&
              "Cannot preprocess this input type!");
     }
+    if (Args.hasArg(options::OPT_fsycl) && OutputTy != types::TY_Dependencies &&
+        Args.hasArg(options::OPT_fsycl_use_footer) &&
+        TargetDeviceOffloadKind == Action::OFK_None) {
+      // Performing a host compilation with -fsycl.  Append the integrated
+      // footer to the preprocessed source file.  We then add another
+      // preprocessed step to complete the action chain.
```

Comment:
Another clarification point: is this additional preprocessor call needed simply to facilitate action collapsing? Is it technically a no-op?

Reply:
We require a true preprocess step of the host, then the append.  The additional preprocess step is the 'normal' step that occurs here.  It's functional and also allows for collapsing.  If the user wants a preprocessed file, this step is even more important as without it we would have an incomplete preprocessed file (appended footer needs to be preprocessed)

Reply:
Got it, thanks!